symengine/functions.cpp

```diff
@@ -1551,6 +1552,8 @@ RCP<const Basic> FunctionWrapper::diff(const RCP<const Symbol> &x) const
 Derivative::Derivative(const RCP<const Basic> &arg, const vec_basic &x)
     : arg_{arg}, x_{x}
 {
+    // TODO: use a multi-set for x_ instead of vector
+    std::sort(x_.begin(), x_.end(), RCPBasicKeyLessCmp());
```

Comment:
You don't want to be doing any work in the constructor. All work must happen before you call the constructor, and then you just check this in `is_canonical`. That way you can quickly construct derivatives if you already have the data in the proper form.